src/NUnitFramework/tests/Attributes/TestOrderAttributeTests.cs

```diff
@@ -22,6 +24,51 @@ public void CheckOrderIsCorrect()
             Assert.AreEqual(work.Children[1].Test.Name, "Y_SecondTest");
             Assert.AreEqual(work.Children[2].Test.Name, "Z_ThirdTest");
         }
-        
+
+        [Test]
+        [TestCaseSource(nameof(Cases))]
+        public void CheckClassOrderIsCorrect(List<Type> candidateTypes)
+        {
+            var testSuite = TestBuilder.MakeFixture(candidateTypes);
+
+            var work = TestBuilder.PrepareWorkItem(testSuite, null) 
+                as CompositeWorkItem;
+
+            var fixtureWorkItems = 
+                ((work.Children[0] as CompositeWorkItem)
+                .Children[0] as CompositeWorkItem)
+                .Children;
+
+            Assert.AreEqual(candidateTypes.Count, fixtureWorkItems.Count);
+            for (var i = 1; i < fixtureWorkItems.Count; i++)
+            {
+                var previousTestOrder = int.Parse(fixtureWorkItems[i - 1]
+                    .Test.Properties.Get(PropertyNames.Order).ToString());
+                var currentTestOrder = int.Parse(fixtureWorkItems[i]
+                    .Test.Properties.Get(PropertyNames.Order).ToString());
+
+                Assert.IsTrue(previousTestOrder < currentTestOrder);
+            }
+        }
+
+        private static readonly object[] Cases =
+        {
```

Comment:
I just tried to change the ordering of classes, both in the source code and in the TestCaseSource `Cases` and the test still passes. So I think it would make sense to have the classes "unordered" in these places to be more confident that the code is actually sorting.